htcrystalball/examine.py

```diff
@@ -0,0 +1,296 @@
+        'ram_usage': '------',
+        'sim_jobs': '------',
+        'wall_time_on_idle': 0
+    }
+
+
+def rename_slot_keys(slot: dict) -> dict:
+    renamed = {
+        'node': slot["UtsnameNodename"],
+        'type': slot["SlotType"],
+        'total_slots': slot["TotalSlots"],
+        'cores': slot["TotalSlotCpus"],
+        'disk': slot["TotalSlotDisk"],
+        'ram': slot["TotalSlotMemory"]
+    }
+
+    if slot.get('TotalSlotGPUs', None):
+        renamed['gpus'] = slot['TotalSlotGPUs']
+
+    return renamed
+
+
+def check_slot_by_type(slot: dict, n_cpu: int, ram: float,
+                       job_duration: float, n_jobs: int, slot_type: str,
+                       n_gpu: int = 0) -> (dict, dict):
+    """
+    Checks all dynamic slots if they fit the job.
+
+    Args:
+        slot: The slot to be checked for running the specified job.
+        n_cpu: The number of CPU cores for a single job
+        ram: The amount of RAM for a single job
+        job_duration: The duration for a single job to execute
+        n_jobs: The number of similar jobs to be executed
+        slot_type: The type of slot, allowed {'static', 'dynamic', 'gpu'}
+        n_gpu: Optional. The number of GPU units for a single job
+
+    Returns:
+        A dictionary of the checked slot and a dictionary with the occupancy
+        details of the slot.
+    """
+    if slot_type not in ['static', 'dynamic', 'gpu']:
+        raise ValueError(f'slot_type must be static, dynamic, or gpu, '
+                         f'not {slot_type}')
+
+    preview = default_preview(slot['UtsnameNodename'], slot_type)
+
+    total_cpus = slot['TotalSlotCpus']
+    total_ram = slot['TotalSlotMemory']
+    total_slots = slot['TotalSlots'] if slot_type == 'static' else 1
+    total_gpus = slot['TotalSlotGPUs'] if slot_type == 'gpu' else 0
+    pct_cpus = int(round((n_cpu / total_cpus) * 100, 0))
+    pct_ram = int(round((ram / total_ram) * 100, 0))
+    fits_job = n_cpu <= total_cpus and ram <= total_ram
+
+    if slot_type == 'gpu':
+        pct_gpu = int(round((n_gpu / total_gpus) * 100, 0))
+        fits_job = fits_job and n_gpu <= total_gpus
+
+        if total_gpus >= 1:
+            preview['gpu_usage'] = f'{n_gpu}/{total_gpus} ({pct_gpu})%'
+        else:
+            preview['gpu_usage'] = 'No GPU resource!'
+
+    preview['core_usage'] = f'{n_cpu}/{total_cpus} ({pct_cpus}%)'
+    preview['ram_usage'] = f'{ram:.2f}/{total_ram} GiB ({pct_ram}%)'
+
+    if fits_job:
+        preview['fits'] = 'YES'
+
+        if slot_type == 'dynamic':
+            preview['sim_jobs'] = min(
+                int(total_cpus / n_cpu),
+                int(total_ram / ram)
+            )
+        elif slot_type == 'gpu':
+            preview['sim_jobs'] = min(
+                int(total_gpus / n_gpu),
+                int(total_cpus / n_cpu),
+                int(total_ram / ram)
+            )
+        elif slot_type == 'static':
+            preview['sim_jobs'] = total_cpus
```

Comment:
@nhjjreuter Why do you have in here just the total_cpus for the number of similar jobs? This should be identical to the dynamic slot since the limits for similar jobs are the same (cpu cores and ram). 

Solution: if slot_type in ['dynamic', 'static']:
...

Reply:
Your original code has `preview_node['sim_jobs'] = available_slots`, where `available_slots = slot["TotalSlotCpus"]` in the `check_static_slots()` function (see line 169 in `my_modules/check_slots.py`.

I can make this change, but it may break the tests while functioning correctly.

Reply:
Ah, okay. Then the total_cpus confused me because available_slots = total_cpus is not always correct. Sometimes a single static slot has multiple cpus assigned, so I used the total slot number.